table/tables/partition.go

```diff
@@ -0,0 +1,169 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tables
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/table"
+	"github.com/pingcap/tidb/tablecodec"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/mock"
+	log "github.com/sirupsen/logrus"
+)
+
+// Table, Partition, PartitionTable both implements the table.Table interface.
+var _ table.Table = &Table{}
+var _ table.Table = &Partition{}
+var _ table.Table = &PartitionTable{}
+
+// PartitionTable implements the table.PartitionTable interface.
+var _ table.PartitionTable = &PartitionTable{}
+
+// Partition is a feature from MySQL:
+// See https://dev.mysql.com/doc/refman/8.0/en/partitioning.html
+// A partition table may contain many partitions, each partition has a unique partition
+// id. The underlying representation of a partition and a normal table (a table with no
+// partitions) is basically the same.
+// Partition also implements the table.Table interface.
+type Partition struct {
+	tableCommon
+}
+
+// PartitionTable implements the table.PartitionTable interface.
+// PartitionTable is a table, it contains many Partitions.
+type PartitionTable struct {
```

Comment:
I prefer `PartitionedTable`. MySQL also uses this terminology.
https://dev.mysql.com/doc/refman/8.0/en/partitioning.html
> The PARTITIONS table in the INFORMATION_SCHEMA database provides information about partitions and **partitioned tables**.